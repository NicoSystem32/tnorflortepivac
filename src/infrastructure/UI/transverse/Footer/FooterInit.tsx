import { useState } from 'react'
import { useSelector, contactMessagesSelector } from '../../../selectors'

// Bootstrap
import { Button, Modal } from 'react-bootstrap'
import Container from 'react-bootstrap/Container'
import ListGroup from 'react-bootstrap/ListGroup'

// Images
import { SuperSolidaria, Cavipetrol } from '../../utils/getImages'

// Icons
import {
    IconInstagramSVG,
    IconFacebookSVG,
    IconTwitterSVG,
    IconLinkedinSVG,
} from '../../utils/getIcons'

// Styles
import './footer.scss'

const FooterInit = (): JSX.Element => {
    const [show, setShow] = useState(false)
    const handleClose = (): void => setShow(false)
    const handleShow = (): void => setShow(true)
    const messages = useSelector(contactMessagesSelector)

    return (
        <Container className="content-footer-init">
            <div className="footer-init">
                <div className="content-social-network">
                    <a
                        id="instagram-ft"
                        className="link-social-network"
                        href="https://www.instagram.com/cavipetrol_oficial/"
                        target="_blank"
                        rel="noreferrer"
                    >
                        <img
                            src={IconInstagramSVG}
                            alt="Instagram"
                            className="social-network instagram"
                        />
                    </a>
                    <a
                        id="facebook-ft"
                        className="link-social-network"
                        href="https://www.facebook.com/Cavipetrol"
                        target="_blank"
                        rel="noreferrer"
                    >
                        <img
                            src={IconFacebookSVG}
                            alt="Facebook"
                            className="social-network facebook"
                        />
                    </a>
                    <a
                        id="twitter-ft"
                        className="link-social-network"
                        href="https://twitter.com/Cavipetrol"
                        target="_blank"
                        rel="noreferrer"
                    >
                        <img
                            src={IconTwitterSVG}
                            alt="Twitter"
                            className="social-network twitter"
                        />
                    </a>
                    <a
                        id="linkedin-ft"
                        className="link-social-network"
                        href="https://www.linkedin.com/company/fondo-de-empleados-cavipetrol/"
                        target="_blank"
                        rel="noreferrer"
                    >
                        <img
                            src={IconLinkedinSVG}
                            alt="Linkedin"
                            className="social-network linkedin"
                        />
                    </a>
                </div>
                <hr className="hr-footer"></hr>
                <div className="title-footer only-desk-footer">Contacto</div>
                <Button variant="primary" onClick={handleShow} className="title-footer-mob">
                    Contáctanos
                </Button>
                <div className="content-firts-info only-desk-footer">
                    <ListGroup variant="flush" className="content-info-footer firts-info">
                        <ListGroup.Item className="info-footer">
                            {messages.find((m) => m.name === 'telefonoBarrancabermeja')?.title}:{' '}
                            {messages.find((m) => m.name === 'telefonoBarrancabermeja')?.text}
                        </ListGroup.Item>
                        <ListGroup.Item className="info-footer">
                            {messages.find((m) => m.name === 'telefonoBogota')?.title}:{' '}
                            {messages.find((m) => m.name === 'telefonoBogota')?.text}
                        </ListGroup.Item>
                        <ListGroup.Item className="info-footer">
                            {messages.find((m) => m.name === 'telefonoBucaramanga')?.title}:{' '}
                            {messages.find((m) => m.name === 'telefonoBucaramanga')?.text}
                        </ListGroup.Item>
                        <ListGroup.Item className="info-footer">
                            {messages.find((m) => m.name === 'telefonoCali')?.title}:{' '}
                            {messages.find((m) => m.name === 'telefonoCali')?.text}
                        </ListGroup.Item>
                        <ListGroup.Item className="info-footer">
                            {messages.find((m) => m.name === 'telefonoCartagena')?.title}:{' '}
                            {messages.find((m) => m.name === 'telefonoCartagena')?.text}
                        </ListGroup.Item>
                        <ListGroup.Item className="info-footer">
                            {messages.find((m) => m.name === 'telefonoCucuta')?.title}:{' '}
                            {messages.find((m) => m.name === 'telefonoCucuta')?.text}
                        </ListGroup.Item>
                    </ListGroup>
                </div>
                <div className="content-second-info only-desk-footer">
                    <ListGroup variant="flush" className="content-info-footer second-info">
                        <ListGroup.Item className="info-footer">
                            {messages.find((m) => m.name === 'telefonoMedellin')?.title}:{' '}
                            {messages.find((m) => m.name === 'telefonoMedellin')?.text}
                        </ListGroup.Item>
                        <ListGroup.Item className="info-footer">
                            {messages.find((m) => m.name === 'telefonoNeiva')?.title}:{' '}
                            {messages.find((m) => m.name === 'telefonoNeiva')?.text}
                        </ListGroup.Item>
                        <ListGroup.Item className="info-footer">
                            {messages.find((m) => m.name === 'telefonoVillavicencio')?.title}:{' '}
                            {messages.find((m) => m.name === 'telefonoVillavicencio')?.text}
                        </ListGroup.Item>
                        <ListGroup.Item className="info-footer">
                            {messages.find((m) => m.name === 'telefonoYopal')?.title}:{' '}
                            {messages.find((m) => m.name === 'telefonoYopal')?.text}
                        </ListGroup.Item>
                        <ListGroup.Item className="info-footer">
                            {messages.find((m) => m.name === 'telefonoOrito')?.title}:{' '}
                            {messages.find((m) => m.name === 'telefonoOrito')?.text}
                        </ListGroup.Item>
                        <ListGroup.Item className="info-footer sub-title-footer">
                            {messages.find((m) => m.name === 'telefonoRestoPais')?.title}:{' '}
                            {messages.find((m) => m.name === 'telefonoRestoPais')?.text}
                        </ListGroup.Item>
                    </ListGroup>
                </div>
                <div className="content-logo-footer">
                    <img alt="Logo Super Solidaria" src={SuperSolidaria} className="logo-super" />
                    <img alt="Logo Cavipetrol" src={Cavipetrol} className="logo-cavipetrol" />
                </div>
                <div className="content-copyright">
                    <p className="firts-copyright">Copyright © 2020 Cavipetrol.</p>
                    <p className="second-copyright"> - </p>
                    <p className="third-copyright">Política de tratamientos personales</p>
                </div>
            </div>
            <Modal show={show} onHide={handleClose} className="modal-contact-footer">
                <Modal.Header closeButton>
                    <Modal.Title className="title-modal-footer">Contáctenos</Modal.Title>
                </Modal.Header>
                <Modal.Body>
                    <ListGroup variant="flush" className="content-modal-footer">
                        <ListGroup.Item className="info-modal-footer">
                            {messages.find((m) => m.name === 'telefonoBarrancabermeja')?.title}:{' '}
                            {messages.find((m) => m.name === 'telefonoBarrancabermeja')?.text}
                        </ListGroup.Item>
                        <ListGroup.Item className="info-modal-footer">
                            {messages.find((m) => m.name === 'telefonoBogota')?.title}:{' '}
                            {messages.find((m) => m.name === 'telefonoBogota')?.text}
                        </ListGroup.Item>
                        <ListGroup.Item className="info-modal-footer">
                            {messages.find((m) => m.name === 'telefonoBucaramanga')?.title}:{' '}
                            {messages.find((m) => m.name === 'telefonoBucaramanga')?.text}
                        </ListGroup.Item>
                        <ListGroup.Item className="info-modal-footer">
                            {messages.find((m) => m.name === 'telefonoCali')?.title}:{' '}
                            {messages.find((m) => m.name === 'telefonoCali')?.text}
                        </ListGroup.Item>
                        <ListGroup.Item className="info-modal-footer">
                            {messages.find((m) => m.name === 'telefonoCartagena')?.title}:{' '}
                            {messages.find((m) => m.name === 'telefonoCartagena')?.text}
                        </ListGroup.Item>
                        <ListGroup.Item className="info-modal-footer">
                            {messages.find((m) => m.name === 'telefonoCucuta')?.title}:{' '}
                            {messages.find((m) => m.name === 'telefonoCucuta')?.text}
                        </ListGroup.Item>
                        <ListGroup.Item className="info-modal-footer">
                            {messages.find((m) => m.name === 'telefonoMedellin')?.title}:{' '}
                            {messages.find((m) => m.name === 'telefonoMedellin')?.text}
                        </ListGroup.Item>
                        <ListGroup.Item className="info-modal-footer">
                            {messages.find((m) => m.name === 'telefonoNeiva')?.title}:{' '}
                            {messages.find((m) => m.name === 'telefonoNeiva')?.text}
                        </ListGroup.Item>
                        <ListGroup.Item className="info-modal-footer">
                            {messages.find((m) => m.name === 'telefonoVillavicencio')?.title}:{' '}
                            {messages.find((m) => m.name === 'telefonoVillavicencio')?.text}
                        </ListGroup.Item>
                        <ListGroup.Item className="info-modal-footer">
                            {messages.find((m) => m.name === 'telefonoYopal')?.title}:{' '}
                            {messages.find((m) => m.name === 'telefonoYopal')?.text}
                        </ListGroup.Item>
                        <ListGroup.Item className="info-modal-footer">
                            {messages.find((m) => m.name === 'telefonoOrito')?.title}:{' '}
                            {messages.find((m) => m.name === 'telefonoOrito')?.text}
                        </ListGroup.Item>
                        <ListGroup.Item className="info-modal-footer-i">
                            {messages.find((m) => m.name === 'telefonoRestoPais')?.title}:{' '}
                            {messages.find((m) => m.name === 'telefonoRestoPais')?.text}
                        </ListGroup.Item>
                    </ListGroup>
                </Modal.Body>
                <Modal.Footer>
                    <Button variant="secondary" onClick={handleClose}>
                        Cerrar
                    </Button>
                </Modal.Footer>
            </Modal>
        </Container>
    )
}

export default FooterInit
