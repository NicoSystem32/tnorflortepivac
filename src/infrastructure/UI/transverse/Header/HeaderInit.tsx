import Container from 'react-bootstrap/Container'
import Navbar from 'react-bootstrap/Navbar'

// Images
import { Logo } from '../../utils/getImages'

// Icons
import { HeadsetSVG, IconUsercogSVG, IconUnlockSVG } from '../../utils/getIcons'

import './header.scss'

const HeaderInit = (): JSX.Element => {
    return (
        <Navbar className="theme-header">
            <Container className="content-header">
                <a id="logo" className="logo" href="/login">
                    <img alt="Logo Cavipetrol" src={Logo} />
                </a>
                <a id="support-hd" className="options-content border-pipe" href="/support">
                    <img src={HeadsetSVG} alt="Servicio al asociado" className="options-img" />
                    <p className="options-text">Servicio al asociado</p>
                </a>
                <a
                    id="forget-password-hd"
                    className="options-content border-pipe"
                    href="/forget-password"
                >
                    <img src={IconUsercogSVG} alt="" className="options-img" />
                    <p className="options-text">Olvidé mi contraseña</p>
                </a>
                <a
                    id="unlocked-user-hd"
                    className="options-content"
                    href="https://transacciones.cavipetrol.com/DwPortalApp/Cavionline/UI/Web/bloqueado_m.aspx?tabID=417&mID=3411"
                    target="_blank"
                    rel="noreferrer"
                >
                    <img
                        alt=""
                        src={IconUnlockSVG}
                        srcSet={IconUnlockSVG}
                        className="options-img"
                    />

                    <p className="options-text">Desbloquear mi usuario</p>
                </a>
                <a
                    id="new-user-hd"
                    className="option-register"
                    href="https://transacciones.cavipetrol.com/DwPortalApp/Cavionline/UI/Web/x_crea0_m.aspx?tabID=417&mID=3411"
                    target="_blank"
                    rel="noreferrer"
                >
                    <span className="register"></span>
                </a>
            </Container>
        </Navbar>
    )
}

export default HeaderInit
