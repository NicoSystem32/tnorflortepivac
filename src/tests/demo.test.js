import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import HeaderInit from '../infrastructure/UI/transverse/Header/HeaderInit'
import FooterInit from '../infrastructure/UI/transverse/Footer/FooterInit'

describe('Pruebas en <HeaderInit />', () => {
    test('debe de mostrar <HeaderInit /> correctamente', () => {
        const wrapper = shallow(<HeaderInit />)

        expect(wrapper).toMatchSnapshot()
    })
})

describe('Pruebas en <FooterInit />', () => {
    test('debe de mostrar <FooterInit /> correctamente', () => {
        const wrapper = shallow(<FooterInit />)

        expect(wrapper).toMatchSnapshot()
    })
})
