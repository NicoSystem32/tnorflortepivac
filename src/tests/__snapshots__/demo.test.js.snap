// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <App /> debe de mostrar <App /> correctamente 1`] = `
<BrowserRouter>
  <div
    className="container mt-5"
  >
    <div
      className="btn-group"
    >
      <Link
        className="btn btn-dark"
        to="/"
      >
        Inicio
      </Link>
      <NavLink
        activeClassName="active"
        className="btn btn-dark"
        to="/login"
      >
        Login
      </NavLink>
    </div>
    <hr />
    <Header />
    <Switch>
      <Route
        path="/login"
      >
        <Login />
      </Route>
      <Route
        exact={true}
        path="/"
      >
        <Home />
      </Route>
    </Switch>
    <Footer />
  </div>
</BrowserRouter>
`;

exports[`Pruebas en <FooterInit /> debe de mostrar <FooterInit /> correctamente 1`] = `
<Container
  className="content-footer-init"
  fluid={false}
>
  <div
    className="footer-init"
  >
    <div
      className="content-social-network"
    >
      <a
        className="link-social-network"
        href="https://www.instagram.com/cavipetrol_oficial/"
        id="instagram-ft"
        target="_blank"
      >
        <img
          alt=""
          className="social-network instagram"
          src="instagram.png"
          srcSet="instagram.svg"
        />
      </a>
      <a
        className="link-social-network"
        href="https://www.facebook.com/Cavipetrol"
        id="facebook-ft"
        target="_blank"
      >
        <img
          alt=""
          className="social-network facebook"
          src="facebook.png"
          srcSet="facebook.svg"
        />
      </a>
      <a
        className="link-social-network"
        href="https://twitter.com/Cavipetrol"
        id="twitter-ft"
        target="_blank"
      >
        <img
          alt=""
          className="social-network twitter"
          src="twitter.png"
          srcSet="twitter.svg"
        />
      </a>
      <a
        className="link-social-network"
        href="https://www.linkedin.com/company/fondo-de-empleados-cavipetrol/"
        id="linkedin-ft"
        target="_blank"
      >
        <img
          alt=""
          className="social-network linkedin"
          src="linkedin.png"
          srcSet="linkedin.svg"
        />
      </a>
    </div>
    <hr
      className="hr-footer"
    />
    <div
      className="title-footer only-desk-footer"
    >
      Contacto
    </div>
    <Button
      active={false}
      className="title-footer-mob"
      disabled={false}
      onClick={[Function]}
      variant="primary"
    >
      Contáctanos
    </Button>
    <div
      className="content-firts-info only-desk-footer"
    >
      <ListGroup
        className="content-info-footer firts-info"
        variant="flush"
      >
        <ListGroupItem
          className="info-footer"
        >
          Barrancabermeja:(60+7) 6973528
        </ListGroupItem>
        <ListGroupItem
          className="info-footer"
        >
          Bogotá: (60+1) 7452322
        </ListGroupItem>
        <ListGroupItem
          className="info-footer"
        >
          Bucaramanga: (60+7) 6973231
        </ListGroupItem>
        <ListGroupItem
          className="info-footer"
        >
          Cali:(60+2) 4893620
        </ListGroupItem>
        <ListGroupItem
          className="info-footer"
        >
          Cartagena:(60+5) 6932471
        </ListGroupItem>
        <ListGroupItem
          className="info-footer"
        >
          Cúcuta:(60+7) 5955919
        </ListGroupItem>
      </ListGroup>
    </div>
    <div
      className="content-second-info only-desk-footer"
    >
      <ListGroup
        className="content-info-footer second-info"
        variant="flush"
      >
        <ListGroupItem
          className="info-footer"
        >
          Medellín:(60+4) 6043238
        </ListGroupItem>
        <ListGroupItem
          className="info-footer"
        >
          Neiva:(60+8) 8630771
        </ListGroupItem>
        <ListGroupItem
          className="info-footer"
        >
          Villavicencio:(60+8) 6784290
        </ListGroupItem>
        <ListGroupItem
          className="info-footer"
        >
          Yopal:(60+8) 6333579
        </ListGroupItem>
        <ListGroupItem
          className="info-footer"
        >
          Orito, Tibú y Rubiales: 3208899981
        </ListGroupItem>
        <ListGroupItem
          className="info-footer sub-title-footer"
        >
          Resto del País: 018000180434
        </ListGroupItem>
      </ListGroup>
    </div>
    <div
      className="content-logo-footer"
    >
      <img
        alt=""
        className="logo-super"
        src="super2x.png"
      />
      <img
        alt=""
        className="logo-cavipetrol"
        src="cavipetrol2x.png"
      />
    </div>
    <div
      className="content-copyright"
    >
      <p
        className="firts-copyright"
      >
        Copyright © 2020 Cavipetrol.
      </p>
      <p
        className="second-copyright"
      >
         - 
      </p>
      <p
        className="third-copyright"
      >
        Política de tratamientos personales
      </p>
    </div>
  </div>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    className="modal-contact-footer"
    dialogAs={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
    enforceFocus={true}
    keyboard={true}
    onHide={[Function]}
    restoreFocus={true}
    show={false}
  >
    <ModalHeader
      closeButton={true}
      closeLabel="Close"
    >
      <ModalTitle
        className="title-modal-footer"
      >
        Contáctenos
      </ModalTitle>
    </ModalHeader>
    <ModalBody>
      <ListGroup
        className="content-modal-footer"
        variant="flush"
      >
        <ListGroupItem
          className="info-modal-footer"
        >
          Barrancabermeja:(60+7) 6973528
        </ListGroupItem>
        <ListGroupItem
          className="info-modal-footer"
        >
          Bogotá: (60+1) 7452322
        </ListGroupItem>
        <ListGroupItem
          className="info-modal-footer"
        >
          Bucaramanga: (60+7) 6973231
        </ListGroupItem>
        <ListGroupItem
          className="info-modal-footer"
        >
          Cali:(60+2) 4893620
        </ListGroupItem>
        <ListGroupItem
          className="info-modal-footer"
        >
          Cartagena:(60+5) 6932471
        </ListGroupItem>
        <ListGroupItem
          className="info-modal-footer"
        >
          Cúcuta:(60+7) 5955919
        </ListGroupItem>
        <ListGroupItem
          className="info-modal-footer"
        >
          Medellín:(60+4) 6043238
        </ListGroupItem>
        <ListGroupItem
          className="info-modal-footer"
        >
          Neiva:(60+8) 8630771
        </ListGroupItem>
        <ListGroupItem
          className="info-modal-footer"
        >
          Villavicencio:(60+8) 6784290
        </ListGroupItem>
        <ListGroupItem
          className="info-modal-footer"
        >
          Yopal:(60+8) 6333579
        </ListGroupItem>
        <ListGroupItem
          className="info-modal-footer"
        >
          Orito, Tibú y Rubiales: 3208899981
        </ListGroupItem>
        <ListGroupItem
          className="info-modal-footer-i"
        >
          Resto del País: 018000180434
        </ListGroupItem>
      </ListGroup>
    </ModalBody>
    <ModalFooter>
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        variant="secondary"
      >
        Cerrar
      </Button>
    </ModalFooter>
  </Modal>
</Container>
`;

exports[`Pruebas en <HeaderInit /> debe de mostrar <HeaderInit /> correctamente 1`] = `
<Fragment>
  <Navbar
    className="theme-header"
    collapseOnSelect={false}
    expand={true}
    variant="light"
  >
    <Container
      className="content-header"
      fluid={false}
    >
      <a
        className="logo"
        href="/login"
        id="logo"
      >
        <img
          alt=""
          src="logo.png"
        />
      </a>
      <a
        className="options-content border-pipe"
        href="/support"
        id="support-hd"
      >
        <img
          alt="Servicio al asociado"
          className="options-img"
          src="headset.svg"
          srcSet="headset.png"
        />
        <p
          className="options-text"
        >
          Servicio al asociado
        </p>
      </a>
      <a
        className="options-content border-pipe"
        href="/forget-password"
        id="forget-password-hd"
      >
        <img
          className="options-img"
          src="users-cog.svg"
          srcSet="users-cog.png"
        />
        <p
          className="options-text"
        >
          Olvidé mi usuario
        </p>
      </a>
      <a
        className="options-content"
        href="https://transacciones.cavipetrol.com/DwPortalApp/Cavionline/UI/Web/bloqueado_m.aspx?tabID=417&mID=3411"
        id="unlocked-user-hd"
        target="_blank"
      >
        <img
          className="options-img"
          src="unlock.svg"
          srcSet="unlock.png"
        />
        <p
          className="options-text"
        >
          Desbloquear mi usuario
        </p>
      </a>
      <a
        className="option-register"
        href="https://transacciones.cavipetrol.com/DwPortalApp/Cavionline/UI/Web/x_crea0_m.aspx?tabID=417&mID=3411"
        id="new-user-hd"
        target="_blank"
      >
        <span
          className="register"
        />
      </a>
    </Container>
  </Navbar>
</Fragment>
`;
